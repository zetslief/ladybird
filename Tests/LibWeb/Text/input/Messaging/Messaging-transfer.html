<script src="../include.js"></script>
<script>
    asyncTest(done => {
        let channel = new MessageChannel();

        channel.port1.onmessage = (event) => {
            println("Port1 received " + event.data.byteLength + " of data"); 
            const array = new Int32Array(event.data);
            println("Final array: " + JSON.stringify(array));
        };

        channel.port2.onmessage = (event) => {
            println("Port2 received " + event.data.byteLength + " of data"); 
            channel.port2.postMessage(event.data, [event.data]);
        };

        // FIXME: These should be output in the order: Port1: A, Port2: A, Port1: B, Port2: B, ...
        const array = new Int32Array([1, 2, 3]);
        channel.port1.start();
        channel.port1.postMessage(array.buffer, [array.buffer]);
        println('Buffer length after transfer: ' + array.buffer.byteLength);
    });
</script>
