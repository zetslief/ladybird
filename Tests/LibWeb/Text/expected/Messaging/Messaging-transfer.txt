Buffer length after transfer: 0
Port2 received undefined of data
Uncaught Error In Test: Uncaught exception: [object DOMException]
